docs/x509/reference.rst

```diff
@@ -1292,12 +1292,42 @@ X.509 CSR (Certificate Signing Request) Builder Object
 
         :return bytes: The DER encoded name.
 
-    .. method:: rfc4514_string()
+    .. method:: rfc4514_string(attr_name_overrides=None)
 
         .. versionadded:: 2.5
+        .. versionchanged:: 36.0
 
-        :return str: Format the given name as a :rfc:`4514` Distinguished Name
-            string, for example ``CN=mydomain.com,O=My Org,C=US``.
+            Added ``attr_name_overrides`` parameter.
+
+        Format the given name as a :rfc:`4514` Distinguished Name
+        string, for example ``CN=mydomain.com,O=My Org,C=US``.
+
+        By default, attributes ``CN``, ``L``, ``ST``, ``O``, ``OU``, ``C``,
+        ``STREET``, ``DC``, ``UID`` are represented by their short name.
+        Unrecognized attributes are formatted as dotted OID strings.
+
+        Example:
+
+        .. doctest::
+
+            >>> name = x509.Name([
+            ...     x509.NameAttribute(NameOID.EMAIL_ADDRESS, "santa@north.pole"),
+            ...     x509.NameAttribute(NameOID.COMMON_NAME, "Santa Claus"),
+            ... ])
```

Comment:
@reaperhulk Should I keep this setup here or move to `testsetup`?

Reply:
It's fine here.